python-frontend/src/main/java/org/sonar/python/semantic/v2/types/TrivialTypeInferenceVisitor.java

```diff
@@ -440,7 +440,7 @@ public void visitName(Name name) {
       .map(Expression.class::cast)
       .map(Expression::typeV2)
       // TODO: classes (SONARPY-1829) and functions should be propagated like other types
-      .filter(t -> (t instanceof ClassType) || (t instanceof FunctionType))
+      .filter(t -> (t instanceof ClassType) || (t instanceof FunctionType) || (t instanceof ModuleType))
```

Comment:
This is needed to ensure we propagate imported names to branches they shouldn't otherwise reach when there is only one of them.
This avoids false negatives.